plugins/hls-alternate-number-format-plugin/src/Ide/Plugin/Literals.hs

```diff
@@ -0,0 +1,146 @@
+
+instance NFData RealSrcSpan where
+    rnf x = x `seq` ()
+
+instance NFData Literal
+
+getSrcText :: Literal -> Text
+getSrcText = \case
+  IntLiteral _ txt _  -> txt
+  FracLiteral _ txt _ -> txt
+
+getSrcSpan :: Literal -> RealSrcSpan
+getSrcSpan = \case
+    IntLiteral ss _ _  -> ss
+    FracLiteral ss _ _ -> ss
+
+-- | Find all literals in a Parsed Source File
+collectLiterals :: (Data ast, Typeable ast) => ast -> [Literal]
+collectLiterals = S.toList . collectLiterals'
+
+collectLiterals' :: (Data ast, Typeable ast) => ast -> Set Literal
+collectLiterals' = everything (<>) (mkQ2 (S.empty :: Set Literal)  traverseLExpr traverseLPat)
+
+-- Located Patterns for whatever reason don't get picked up when using `(mkQ (S.empty :: Set Literal) traverseLExpr)
+-- as such we need to explicit traverse those in order to pull out any literals
+mkQ2 :: (Typeable a, Typeable b, Typeable c) => r -> (b -> r) -> (c -> r) -> a -> r
+mkQ2 def left right datum = case cast datum of
+        Just datum' -> left datum'
+        Nothing     -> maybe def right (cast datum)
+
+traverseLPat :: GenLocated SrcSpan (Pat GhcPs) -> Set Literal
+traverseLPat (L sSpan pat) = traversePat sSpan pat
+
+traversePat :: SrcSpan -> Pat GhcPs -> Set Literal
+traversePat sSpan = \case
+    LitPat _ lit                             -> getLiteralAsList sSpan lit
+    NPat _ (L olSpan overLit) sexpr1 sexpr2  -> getOverLiteralAsList olSpan overLit
+                                            <> collectLiterals' sexpr1
+                                            <> collectLiterals' sexpr2
+    NPlusKPat _ _ (L olSpan loverLit) overLit sexpr1 sexpr2 -> getOverLiteralAsList olSpan loverLit
+                                                    <> getOverLiteralAsList sSpan overLit
+                                                    <> collectLiterals' sexpr1
+                                                    <> collectLiterals' sexpr2
+    ast -> collectLiterals' ast
+
+traverseLExpr :: GenLocated SrcSpan (HsExpr GhcPs) -> Set Literal
+traverseLExpr (L sSpan hsExpr) = traverseExpr sSpan hsExpr
+
+traverseExpr :: SrcSpan -> HsExpr GhcPs -> Set Literal
+traverseExpr sSpan = \case
+      HsOverLit _ overLit -> getOverLiteralAsList sSpan overLit
+      HsLit _ lit         -> getLiteralAsList sSpan lit
+      expr                -> collectLiterals' expr
+
+getLiteralAsList :: SrcSpan -> HsLit GhcPs -> Set Literal
+getLiteralAsList sSpan lit =  case sSpan of
+    RealSrcSpan rss _ -> getLiteralAsList' lit rss
+    _                 -> S.empty
+
+getLiteralAsList' :: HsLit GhcPs -> RealSrcSpan -> Set Literal
+getLiteralAsList' lit = maybe S.empty S.singleton . flip getLiteral lit
+
+-- Translate from Hs Type to our Literal type
+getLiteral :: RealSrcSpan -> HsLit GhcPs -> Maybe Literal
+getLiteral sSpan = \case
+  HsInt _ val   -> fromIntegralLit sSpan val
+-- Ignore this case for now
+--   HsInteger _ val _           -> Just $ IntLiteral sSpan Nothing val
```

Comment:
What is this case, and why is it ignored?

Reply:
I never found `HsInteger` in any of the test files I used. I didn't want to lose note of it just in case I ran into it.

Reply:
Removed.